miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -371,6 +378,37 @@ private void validateAliasUniqueness(Sample sample) throws IOException {
     }
   }
 
+  /**
+   * Adds any external names in the sample's attached identity to the identity stored in the database
+   * 
+   * @param sample
+   * @throws IOException
+   */
+
+  private void addExternalNames(DetailedSample sample, SampleIdentity identityCopy) throws IOException {
+    if (identityCopy == null || identityCopy.getExternalName() == null) return;
+    SampleIdentity identity = (SampleIdentity) get(getIdentity(sample).getId());
+    Set<String> identityExternalNames = SampleIdentityImpl.getSetFromString(identity.getExternalName());
+    Set<String> tempExternalNames = SampleIdentityImpl.getSetFromString(identityCopy.getExternalName());
+    Set<String> lowerCaseIdentityExternalNames = identityExternalNames.stream().map(String::toLowerCase).collect(Collectors.toSet());
+    tempExternalNames.stream().forEach(name -> {
+      try {
+        if (!lowerCaseIdentityExternalNames.contains(name.toLowerCase()) && !(isUniqueExternalNameWithinProjectRequired() &&
+            (identity.getProject() == null
+                || getIdentitiesByExternalNameOrAliasAndProject(name, identity.getProject().getId(), true).size() > 0))) {
+          identityExternalNames.add(name);
+        }
+      } catch (IOException e) {
+        log.error("Failed to retrieve all external names: " + e);
+      }
+    });
+
+    identity.setExternalName(String.join(",", identityExternalNames));
+    if (identityExternalNames.size() > lowerCaseIdentityExternalNames.size()) {
+      sampleStore.update(identity);
```

Comment:
Please call `setChangeDetails(identity)` before the save, so the lastModified and lastModifier values get set.